src/xfer/cache.rs

```diff
@@ -25,38 +32,56 @@ pub enum CacheError {
 }
 
 #[derive(Debug)]
-pub struct Cache {
+pub struct Cache<State> {
     pub(crate) settings: CacheSettings,
+    state: PhantomData<State>,
```

Comment:
Both states are ZSTs anyway, so there's no need to wrap them in `PhantomData`.

Reply:
I think otherwise the struct complains about `State` not being used. You can avoid using `PhantomData` when the struct uses the generic type (which we don't need / do here).

Reply:
Renaming the field to begin with `_` should also silence that warning (`_state`).